Zend/zend_compile.h

```diff
@@ -299,6 +299,9 @@ typedef struct _zend_property_info {
 	zend_string *name;
 	zend_string *doc_comment;
 	zend_class_entry *ce;
+	zend_uchar type;
+	zend_string *type_name;
```

Comment:
shouldn't these be a union? You can either have a scalar/array type or a class entry, not both.